src/find-options/FindOptionsUtils.ts

```diff
@@ -94,6 +96,11 @@ export class FindOptionsUtils {
             return qb;
 
         // apply all options from FindOptions
+        if (options.select) {
+            qb.select(options.select.map(selection => qb.alias + "." + selection));
+            qb.partialSelect = true;
```

Comment:
> because a primary column is inserted into the sql

you need to insert primary columns only if they aren't selected at all (if something is selected, but not primary columns). You don't need this ugly flag. Try to redesign this thing